packages/pages/src/upgrade/pagesUpdater.ts

```diff
@@ -407,3 +407,138 @@ export const checkNodeVersion = () => {
     console.log(ERR_MSG);
   }
 };
+
+const oldServerlessFunctionRequestTypes = [
+  "SitesHttpRequest",
+  "SitesOnUrlChangeRequest",
+];
+
+const oldServerlessFunctionResponseTypes = [
+  "SitesHttpResponse",
+  "Promise<SitesHttpResponse>",
+  "SitesOnUrlChangeResponse",
+  "Promise<SitesOnUrlChangeResponse>",
+];
+
+const oldServerlessFunctionTypes = [
+  ...oldServerlessFunctionRequestTypes,
+  ...oldServerlessFunctionResponseTypes,
+];
+
+/**
+ * Updates the imports and usages of e.g. SitesHttpRequest to PagessHttpRequest.
+ * @param serverlessFunctionsPath the path to the serverless functions folder
+ */
+export const updateServerlessFunctionTypeReferences = (
+  serverlessFunctionsPath: string
+) => {
+  let updated = false;
+  const project = new Project({
+    compilerOptions: {
+      jsx: typescript.JsxEmit.ReactJSX,
+      sourceRoot: serverlessFunctionsPath,
+    },
+  });
+  project.addSourceFilesAtPaths(
+    `${serverlessFunctionsPath}/**/*.{ts,tsx,js,jsx}`
+  );
+  const sourceFiles = project.getSourceFiles();
+  for (let i = 0; i < sourceFiles.length; i++) {
+    let fileUpdated = false;
+    const sourceFile = sourceFiles[i];
+    const importDeclarations = sourceFile.getImportDeclarations();
+    for (let j = 0; j < importDeclarations.length; j++) {
+      const importDeclaration = importDeclarations[j];
+      if (importDeclaration.getModuleSpecifierValue() !== "@yext/pages/*") {
+        continue;
+      }
+      const namedImports = importDeclaration.getNamedImports();
+      for (let k = 0; k < namedImports.length; k++) {
+        const namedImport = namedImports[k];
+        if (!oldServerlessFunctionTypes.includes(namedImport.getName())) {
+          continue;
+        }
+
+        switch (namedImport.getName()) {
+          case "SitesHttpRequest":
```

Comment:
I think a map would be a more appropriate structure here, then you could have a map of 'SitesHttpResponse' -> 'PagesHttpResponse' instead of needing a switch case for each replacement.